src/main/java/org/team1540/robot2025/subsystems/intake/IntakeIOReal.java

```diff
@@ -0,0 +1,122 @@
+
+public class IntakeIOReal implements IntakeIO {
+
+    // rotation of horizontal beams for the intake, clockwise to intake, counter-clockwise to spit out
+    private final TalonFX spinFalcon = new TalonFX(IntakeConstants.SPIN_FALCON_ID);
+
+    // controls position of intake
+    private final TalonFX pivotFalcon = new TalonFX(IntakeConstants.PIVOT_FALCON_ID);
+
+    private final MotionMagicVoltage pivotPositionRequest = new MotionMagicVoltage(0).withSlot(0);
+
+    // clockwise to intake, counter-clockwise to spit out
+    private final SparkMax funnelNeo = new SparkMax(IntakeConstants.NEO_ID, SparkLowLevel.MotorType.kBrushless);
+
+    private StatusCode spinFalconStatus;
+
+    private StatusCode pivotFalconStatus;
+
+    public IntakeIOReal() {
+
+        TalonFXConfiguration spinTalonFXConfigs = new TalonFXConfiguration();
+        TalonFXConfiguration pivotTalonFXConfigs = new TalonFXConfiguration();
+        SparkMaxConfig funnelNEOConfig = new SparkMaxConfig();
+
+        spinTalonFXConfigs.CurrentLimits.withStatorCurrentLimitEnable(true);
+        spinTalonFXConfigs.CurrentLimits.withStatorCurrentLimit(120);
+        spinTalonFXConfigs.CurrentLimits.withSupplyCurrentLimit(70);
+        spinTalonFXConfigs.CurrentLimits.withSupplyCurrentLowerLimit(40);
+        spinTalonFXConfigs.CurrentLimits.withSupplyCurrentLowerTime(0.5);
+        spinTalonFXConfigs.MotorOutput.Inverted = InvertedValue.Clockwise_Positive;
+        spinTalonFXConfigs.MotorOutput.NeutralMode = NeutralModeValue.Coast;
+
+        Slot0Configs spinConfigs = spinTalonFXConfigs.Slot0;
+
+        spinFalcon.getConfigurator().apply(spinConfigs);
+
+        pivotTalonFXConfigs.CurrentLimits.withStatorCurrentLimitEnable(true);
+        pivotTalonFXConfigs.CurrentLimits.withStatorCurrentLimit(120);
+        pivotTalonFXConfigs.CurrentLimits.withSupplyCurrentLimit(70);
+        pivotTalonFXConfigs.CurrentLimits.withSupplyCurrentLowerLimit(40);
+        pivotTalonFXConfigs.CurrentLimits.withSupplyCurrentLowerTime(0.5);
+        pivotTalonFXConfigs.MotorOutput.Inverted = InvertedValue.CounterClockwise_Positive;
+        pivotTalonFXConfigs.MotorOutput.NeutralMode = NeutralModeValue.Brake;
+
+        Slot0Configs pivotConfigs = pivotTalonFXConfigs.Slot0;
+
+        pivotConfigs.kS = IntakeConstants.PIVOT_INTAKE_KS;
+        pivotConfigs.kV = IntakeConstants.PIVOT_INTAKE_KV;
+        pivotConfigs.kA = IntakeConstants.PIVOT_INTAKE_KA;
+        pivotConfigs.kP = IntakeConstants.PIVOT_INTAKE_KP;
+        pivotConfigs.kI = IntakeConstants.PIVOT_INTAKE_KI;
+        pivotConfigs.kD = IntakeConstants.PIVOT_INTAKE_KD;
+
+        MotionMagicConfigs motionMagicConfigs = pivotTalonFXConfigs.MotionMagic;
+
+        motionMagicConfigs.MotionMagicCruiseVelocity = IntakeConstants.PIVOT_INTAKE_CRUISE_VELOCITY;
+        motionMagicConfigs.MotionMagicAcceleration = IntakeConstants.PIVOT_INTAKE_ACCELERATION;
+
+        pivotFalcon.getConfigurator().apply(pivotConfigs);
```

Comment:
Same as above. This will only apply the PID gain configs and nothing else. Apply `pivotTalonFXConfigs`, which is a full `TalonFXConfiguration` object, and not `pivotConfigs`, which only contains PID gains. And also, make your variable names clearer so this confusion doesn't happen again in the future.